sdk/src/Services/AutoScaling/Generated/Internal/AutoScalingAuthSchemeResolver.cs

```diff
@@ -0,0 +1,63 @@
+using Amazon.Runtime;
+using Amazon.Runtime.Credentials.Internal;
+using Amazon.Runtime.Internal;
+using Smithy.Identity.Abstractions;
+using System.Collections.Generic;
+
+// Hand-written class at the moment, generated files will include documentation for public members.
+#pragma warning disable CS1591
+
+namespace Amazon.AutoScaling.Internal
+{
+    // Auth parameters and scheme resolvers will be generated for each service;
+    // The logic for S3 and EventBridge (where authentication rules are in the endpoint-rule-set file) will be different.
+
+    public class AutoScalingAuthSchemeParameters : IAuthSchemeParameters
+    {
+        public string Operation { get; set; }
+        public string Region { get; set; }
+    }
+
+    public class AutoScalingAuthSchemeResolver : 
```

Comment:
This class is a both an SDK pipeline handler and a Smithy RA auth resolver which I think is a bit odd. The class should be one thing not both. I would suggest creating `AutoScalingAuthSchemeHandler` that would be generated to convert the `IRequestContext` into `AutoScalingAuthSchemeParameters`. Then have the handler call the `AutoScalingAuthSchemeResolver`.